protocols/kad/src/handler.rs

```diff
@@ -368,6 +365,9 @@ pub enum KademliaHandlerIn {
     /// for the query on the remote.
     Reset(KademliaRequestId),
 
+    /// Change the connection to the specified mode.
```

Comment:
What's the effect of that? Is there a network message sent all my connections when I learn about external_addresses? Isn't that a lot of overhead? 

What if my addresses go on and off often? Not sure why that might happen, but now how do I make sure that doesn't happen?

Reply:
Nobody receives any messages. Client vs server mode essentially just means, "do we accept inbound kademlia streams" which translates to `rust-libp2p` as "do we offer kademlia via `ConnectionHandler::listen_protocols`".